background_task/models.py

```diff
@@ -254,10 +254,13 @@ def reschedule(self, type, err, traceback):
             task_failed.send(sender=self.__class__, task_id=self.id, completed_task=completed)
             self.delete()
         else:
-            backoff = timedelta(seconds=(self.attempts ** 4) + 5)
-            self.run_at = timezone.now() + backoff
-            logger.warning('Rescheduling task %s for %s later at %s', self,
-                backoff, self.run_at)
+            delay_settings = app_settings.BACKGROUND_TASK_DELAY_BETWEEN_ATTEMPTS
+            backoff = delay_settings(self.attempts) if callable(delay_settings) else delay_settings
+            assert isinstance(backoff, int), \
+                "BACKGROUND_TASK_DELAY_BETWEEN_ATTEMPTS must be an integer or a callable that returns an integer. " \
+                "Got %s (%s)." % (backoff, type(backoff).__name__)
```

Comment:
It is convenient to specify the type of the invalid setting or result of the callable, because a string value in settings (like '1') would result in the following message:

```python
"BACKGROUND_TASK_DELAY_BETWEEN_ATTEMPTS must be an integer or a callable that returns an integer. Got 1."  # '1' would not be single quoted in that case.
```

Better get an explicit:

```python
"BACKGROUND_TASK_DELAY_BETWEEN_ATTEMPTS must be an integer or a callable that returns an integer. Got 1 (str)."
```

... hence the need to avoid shadowing the `type` built-in function in the parameter.